packages/loader/driver-utils/src/runWithRetry.ts

```diff
@@ -67,16 +70,20 @@ export async function runWithRetry<T>(
             }
 
             if (progress.cancel?.aborted === true) {
+                const cancelDescription = progress.cancelDescription;
```

Comment:
Why not use AbortSignal.reason?
See
https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/reason
https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort

Reply:
reason doesn't exist in all environments we run

Reply:
I'd rather add reason (and if needed, did appropriate casting to make it compile in Node14) then invent our own mechanisms. We would lose it in Node14 environment - not great, but if Node16 is on horizon, then it feels like better flow than coming up with alternative flow, especially that it does not solve the problem - it does not tell why container was closed.

Reply:
I didn't see any "reason" in my intellisense for abort or AbortSignal so that's why I rolled my own. Was wishing it was there so glad to hear it is. I'll have to take a look at why I don't see it. Maybe out of date types packages?

Reply:
@anthony-murphy , I agree with you (exceptions for exceptional cases, aka not expected code flows / errors), but I think it's unrelated to problem in hand. We can convert Container.attach() to return "aborted" state (instead of exception), with a bit better reason (container is closed), but in principle it does not change anything. We can (with exactly same result, at least for now), check if container is closed and not rethrow this error in Container.attach() and return something else. I.e. we can easily decouple generic flow to exceptions being for exceptional cases to how we deal with .attach() in closed container case.

Reply:
my feedback is more around i would love to see us start moving away from expections. this helper function seems like a super easy place to begin that journey, and to have an example for others to follow. it will take time to get it used everywhere

Reply:
Switching to Draft to try it out.

Reply:
Resolution here:

* Added new function that returns a Result type instead of throwing
* The "aborted" case has no "reason", since we don't have full support - both from JS runtime (Node 14) and lib.dom.d.ts typings (coming in TS 4.7)
* We can still get a sufficiently detailed error message since Container.attach can craft it itself.